File: backend/dto/coach.go
package dto

type CoachMoveRequest struct {
	FenBefore        string             `json:"fen_before"`
	UserMove         string             `json:"user_move"`
	FenAfter         string             `json:"fen_after" binding:"required"`
	Elo              int                `json:"elo"`
	History          []string           `json:"history"`
	RecentUserMoves  []RecentUserMove   `json:"recent_user_moves"`
}

type RecentUserMove struct {
	San      string `json:"san"`
	Category string `json:"category"`
	CpLoss   int    `json:"cp_loss"`
}

type CoachUserMove struct {
	Uci      string `json:"uci"`
	Category string `json:"category"`
	CpLoss   int    `json:"cp_loss"`
	BestMove string `json:"best_move"`
	ScoreCp  *int   `json:"score_cp"`
	Mate     *int   `json:"mate"`
}



type CoachBotMove struct {
	Uci string `json:"uci"`
}

type CoachMoveResponse struct {
	UserMove     *CoachUserMove `json:"user_move"`
	BotMove      *CoachBotMove  `json:"bot_move"`
	FenAfterBot  string         `json:"fen_after_bot"`
	GameOver     bool           `json:"game_over"`
	Result       *string        `json:"result"`
	CoachMessage *string        `json:"coach_message"`
}

File: frontend/src/services/coach.js
import api from "./api";

export async function coachMove({
    fenBefore,
    userMove,
    fenAfter,
    elo,
    history = [],
    recentUserMoves = [],
}) {
    const { data } = await api.post(
        "/api/coach/move",
        {
            fen_before: fenBefore,
            user_move: userMove,
            fen_after: fenAfter,
            elo,
            history,
            recent_user_moves: recentUserMoves,
        },
        { timeout: 30 * 1000 },
    );
    return data;
}
